src/parser/globals.ts

```diff
@@ -0,0 +1,17 @@
+import { VERSION as SVELTE_VERSION } from "svelte/compiler";
+
+const globalsForSvelte4: Readonly<string[]> = [
+  "$$slots",
+  "$$props",
+  "$$restProps",
+] as const;
+export const globalsForSvelte5 = [
+  "$state",
+  "$derived",
+  "$effect",
+  "$effect.pre",
```

Comment:
Since `$effect.pre` is a member of `$effect`, I don't think it is necessary to define it here.

```suggestion
```

Reply:
Fixed!